query-engine/connectors/sql-query-connector/src/database/operations/read.rs

```diff
@@ -141,13 +143,79 @@ pub async fn get_related_m2m_record_ids(
         .collect())
 }
 
-pub async fn count_by_model(
+pub async fn aggregate(
     conn: &dyn QueryExt,
     model: &ModelRef,
+    aggregators: Vec<Aggregator>,
     query_arguments: QueryArguments,
-) -> crate::Result<usize> {
-    let query = read::count_by_model(model, query_arguments);
-    let count = conn.find_int(query).await? as usize;
+) -> crate::Result<Vec<AggregationResult>> {
+    let query = read::aggregate(model, &aggregators, query_arguments);
+    let idents = extract_query_idents(&aggregators);
 
-    Ok(count)
+    let mut rows = conn.filter(query.into(), idents.as_slice()).await?;
+    let row = rows
+        .pop()
+        .expect("Expected exactly one return row for aggregation query.");
+
+    Ok(parse_aggregation_row(&aggregators, row))
+}
+
+fn extract_query_idents(aggregators: &[Aggregator]) -> Vec<(TypeIdentifier, FieldArity)> {
+    aggregators
+        .iter()
+        .flat_map(|aggregator| match aggregator {
+            Aggregator::Count => vec![(TypeIdentifier::Int, FieldArity::Required)],
+            Aggregator::Average(fields) => map_aggregator_field_types(&fields, Some(TypeIdentifier::Float)),
+            Aggregator::Sum(fields) => map_aggregator_field_types(&fields, None),
+            Aggregator::Min(fields) => map_aggregator_field_types(&fields, None),
+            Aggregator::Max(fields) => map_aggregator_field_types(&fields, None),
+        })
+        .collect()
+}
+
+fn map_aggregator_field_types(
+    fields: &[ScalarFieldRef],
+    fixed_type: Option<TypeIdentifier>,
+) -> Vec<(TypeIdentifier, FieldArity)> {
+    fields
+        .into_iter()
+        .map(|f| {
+            (
+                fixed_type.clone().unwrap_or(f.type_identifier.clone()),
+                FieldArity::Required,
+            )
+        })
+        .collect()
+}
+
+fn parse_aggregation_row(aggregators: &[Aggregator], row: SqlRow) -> Vec<AggregationResult> {
```

Comment:
This might be nicer in `SqlRow`:

```rust
impl SqlRow {
    pub fn parse_aggregation(self, aggregators: &[aggregator]) -> Vec<AggregationResult> {
        let mut values = self.values;
        values.reverse();

        aggregators
            .iter()
            .flat_map(|aggregator| match aggregator {
                Aggregator::Count => vec![AggregationResult::Count(values.pop().unwrap())],

                Aggregator::Average(fields) => fields
                    .iter()
                    .map(|field| AggregationResult::Average(field.clone(), values.pop().unwrap()))
                    .collect(),

                Aggregator::Sum(fields) => fields
                    .iter()
                    .map(|field| AggregationResult::Sum(field.clone(), values.pop().unwrap()))
                    .collect(),

                Aggregator::Min(fields) => fields
                    .iter()
                    .map(|field| AggregationResult::Min(field.clone(), values.pop().unwrap()))
                    .collect(),

                Aggregator::Max(fields) => fields
                    .iter()
                    .map(|field| AggregationResult::Max(field.clone(), values.pop().unwrap()))
                    .collect(),
            })
            .collect()
    }
}
```